### PR TITLE
fix: deref document even when you arent using CLI

### PR DESCRIPTION
**Pull request by BelfordZ**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -98,8 +99,8 @@ const languageFilenameMap: any = {
 
 export default async (generatorOptions: IGeneratorOptions) => {
   const { openrpcDocument, outDir } = generatorOptions;
-
-  const methodTypings = new MethodTypings(openrpcDocument);
+  const dereffed = await parseOpenRPCDocument(openrpcDocument);
```
Comment (shanejonas) [inline src/index.ts]: wouldnt this change make it deref twice on `cli` usage?

Comment (BelfordZ) [reply src/index.ts]: fixed

File: src/index.ts [hunk 2]
```diff
@@ -98,8 +99,17 @@ const languageFilenameMap: any = {
 
 export default async (generatorOptions: IGeneratorOptions) => {
   const { openrpcDocument, outDir } = generatorOptions;
-
-  const methodTypings = new MethodTypings(openrpcDocument);
+  let dereffedDocument;
+  
+  try {
+    dereffedDocument = await parseOpenRPCDocument(program.document);
```
Comment (BelfordZ) [inline src/index.ts]:
```suggestion
    dereffedDocument = await parseOpenRPCDocument(openrpcDocument);
```

File: src/cli.ts [hunk 3]
```diff
@@ -32,16 +31,9 @@ program
   .action(async () => {
     let openrpcDocument: OpenRPC;
```
Comment (BelfordZ) [inline src/cli.ts]:
```suggestion
```

File: src/index.ts [hunk 4]
```diff
@@ -98,8 +99,17 @@ const languageFilenameMap: any = {
 
 export default async (generatorOptions: IGeneratorOptions) => {
   const { openrpcDocument, outDir } = generatorOptions;
+  let dereffedDocument;
 
-  const methodTypings = new MethodTypings(openrpcDocument);
+  try {
+    dereffedDocument = await parseOpenRPCDocument(openrpcDocument);
+  } catch (e) {
+    console.error(e.message); // tslint:disable-line
```
[the remark this answers is not in the document]
Comment (BelfordZ) [reply src/index.ts]: does one EVER need to ts lint? lol

**Pull request merged**